### ISSUE TITLE
Add better support for managing client connections.

### ISSUE THREAD

**Issue by mvanderlee** [alekc/terraform-provider-auth0#21]
### Description

Auth0 links clients and connections on the connection object, which makes it very difficult to add and remove a single client from a connection. 
We should add a way to link a client and connection. Something similar to AWS's role policy attachment: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy_attachment

### New or Affected Resource(s)

* auth0_client_connection

### Potential Terraform Configuration

```hcl
resource "auth0_client_connection" "this" {
  client_id = "obvious_fake"
  connection_id = "con_fake"
}
```

### References

* https://community.auth0.com/t/how-to-create-an-client-with-a-single-connection/37876

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by btilford** [comment]
Would this solve the scenario where you have internal clients managed with Terraform and external ones managed through another application which results with any client not being managed by Terraform being removed from the connection every time Terraform runs?

**Comment by mvanderlee** [comment]
@btilford yes. I've already used the source branch in PR#22  in production to do exactly that.
